File: 0977 squares-of-a-sorted-array/solution.java
class Solution {
    public int[] sortedSquares(int[] nums) {
     int n=nums.length;
     int []res=new int[n];
     int l=0,r=n-1;
     for(int i=n-1;i>=0;i--) {
        int sq1=nums[l]*nums[l];
        int sq2=nums[r]*nums[r];
        if(sq1>sq2) {
            res[i]=sq1;
            l++;
        }
        else {
            res[i]=sq2;
            r--;
        }
     }
     return res;
    }
}

File: 0977 squares-of-a-sorted-array/solution.go
func sortedSquares(nums []int) []int {
    n:=len(nums)
    var arr = make([]int,n)
    l,r:=0,n-1

    for i:=n-1;i>=0;i-- {
        sq1:=nums[l]*nums[l]
        sq2:=nums[r]*nums[r];
        if sq1>sq2 {
            arr[i]=sq1
            l++
        } else {
            arr[i]=sq2
            r--
        }
    }
    return arr
}
